internal/app/machined/internal/platform/baremetal/baremetal.go

```diff
@@ -88,9 +91,27 @@ func (b *BareMetal) Initialize(data *userdata.UserData) (err error) {
 		return err
 	}
 
-	i := installer.NewInstaller(cmdline, data)
-	if err = i.Install(); err != nil {
-		return errors.Wrap(err, "failed to install")
+	// Attempt to discover a previous installation
+	// An err case should only happen if no partitions
+	// with matching labels were found
+	var mountpoints *mount.Points
+	mountpoints, err = owned.MountPointsFromLabels(true)
+	if err != nil {
```

Comment:
I don't think determining if an installation has occurred from this error is the right route. The probing logic is complex and it could really be any number of issues that causes the probing to fail.

Reply:
decision was based on the error being thrown here. 
https://github.com/talos-systems/talos/blob/master/internal/pkg/mount/manager/owned/owned.go#L68

It seemed reasonable that if probe stuff failed for a more specific ( bad disk? ) reason that it'd fail during the installation process when we try to read/write to the disk.